numba/npyufunc/parallel.py

```diff
@@ -450,12 +456,134 @@ def raise_with_hint(required):
+    ll.add_symbol('get_thread_num', lib.get_thread_num)
+
+    global _set_num_threads
+    _set_num_threads = CFUNCTYPE(None, c_int)(lib.set_num_threads)
+    _set_num_threads(NUM_THREADS)
+
+    global _get_num_threads
+    _get_num_threads = CFUNCTYPE(c_int)(lib.get_num_threads)
+
+    global _get_thread_num
+    _get_thread_num = CFUNCTYPE(c_int)(lib.get_thread_num)
+
+
+# Some helpers to make set_num_threads jittable
+
+
+def snt_check(n):
+    from numba.config import NUMBA_NUM_THREADS
+    msg = "The number of threads must be between 1 and %s" % NUMBA_NUM_THREADS
+    if n > NUMBA_NUM_THREADS or n < 1:
+        raise ValueError(msg)
+
+
+@overload(snt_check)
+def ol_snt_check(n):
+    from numba.config import NUMBA_NUM_THREADS
+    msg = "The number of threads must be between 1 and %s" % NUMBA_NUM_THREADS
+
+    def impl(n):
+        if n > NUMBA_NUM_THREADS or n < 1:
+            raise ValueError(msg)
+    return impl
+
+
+def set_num_threads(n):
+    """
+    Set the number of threads to use for parallel execution.
+
+    By default, all :obj:`numba.config.NUMBA_NUM_THREADS` threads are used.
+
+    This functionality works by masking out threads that are not used.
+    Therefore, the number of threads *n* must be less than or equal to
+    :obj:`~.NUMBA_NUM_THREADS`, the total number of threads that are launched.
+    See its documentation for more details.
+
+    This function can be used inside of a jitted function.
+
+    Parameters
+    ----------
+    n: The number of threads. Must be between 1 and NUMBA_NUM_THREADS.
+
+    See Also
+    --------
+    get_num_threads, numba.config.NUMBA_NUM_THREADS,
+    numba.config.NUMBA_DEFAULT_NUM_THREADS, :envvar:`NUMBA_NUM_THREADS`
+
+    """
+    _launch_threads()
+    snt_check(n)
+    _set_num_threads(n)
+
+
+@overload(set_num_threads)
+def ol_set_num_threads(n):
+    _launch_threads()
+    def impl(n):
+        snt_check(n)
+        _set_num_threads(n)
+    return impl
+
+
+def get_num_threads():
+    """
+    Get the number of threads used for parallel execution.
+
+    By default (if :func:`~.set_num_threads` is never called), all
+    :obj:`numba.config.NUMBA_NUM_THREADS` threads are used.
+
+    This number is less than or equal to the total number of threads that are
+    launched, :obj:`numba.config.NUMBA_NUM_THREADS`.
+
+    This function can be used inside of a jitted function.
+
+    Returns
+    -------
+    The number of threads.
+
+    See Also
+    --------
+    set_num_threads, numba.config.NUMBA_NUM_THREADS,
+    numba.config.NUMBA_DEFAULT_NUM_THREADS, :envvar:`NUMBA_NUM_THREADS`
+
+    """
+    _launch_threads()
+    return _get_num_threads()
+
+
+@overload(get_num_threads)
+def ol_get_num_threads():
+    _launch_threads()
+    def impl():
+        return _get_num_threads()
+    return impl
+
+def get_thread_num():
+    """
+    docs
+    """
+    return _get_thread_num()
+
+@overload(get_thread_num)
+def ol_get_thread_num():
+    def impl():
```

Comment:
This needs a `_launch_threads()` call in the outer scope of the overload like https://github.com/numba/numba/pull/4615/files#diff-ec5b6bd5010b2b32f80834aad542033cR570

Reply:
This looks to be resolved now?